tests/e2e/svcpoliciesandfirewall/svcpoliciesandfirewall_test.go

```diff
@@ -0,0 +1,363 @@
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Checks Node Status", func() {
+		Eventually(func(g Gomega) {
+			var err error
+			nodes, err = e2e.ParseNodes(kubeConfigFile, false)
+			g.Expect(err).NotTo(HaveOccurred())
+			for _, node := range nodes {
+				g.Expect(node.Status).Should(Equal("Ready"))
+			}
+		}, "300s", "5s").Should(Succeed())
+		_, err := e2e.ParseNodes(kubeConfigFile, true)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Checks Pod Status", func() {
+		Eventually(func(g Gomega) {
+			pods, err := e2e.ParsePods(kubeConfigFile, false)
+			g.Expect(err).NotTo(HaveOccurred())
+			for _, pod := range pods {
+				if strings.Contains(pod.Name, "helm-install") {
+					g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
+				} else {
+					g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
+				}
+			}
+		}, "300s", "5s").Should(Succeed())
+		_, err := e2e.ParsePods(kubeConfigFile, true)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	// Verifies that the service with external traffic policy=local is deployed
+	// Verifies that the external-ip is only set to the node IP where the server runs
+	// It also verifies that the service with external traffic policy=cluster has both node IPs as externalIP
+	It("Verify external traffic policy=local gets set up correctly", func() {
+		_, err := e2e.DeployWorkload("loadbalancer.yaml", kubeConfigFile, *hardened)
+		Expect(err).NotTo(HaveOccurred(), "loadbalancer not deployed")
+		_, err = e2e.DeployWorkload("loadbalancer-extTrafficPol.yaml", kubeConfigFile, *hardened)
+		Expect(err).NotTo(HaveOccurred(), "loadbalancer-extTrafficPol not deployed")
+
+		// Check where the server pod is running
+		var serverNodeName string
+		Eventually(func() (string, error) {
+			pods, err := e2e.ParsePods(kubeConfigFile, false)
+			Expect(err).NotTo(HaveOccurred(), "failed to parse pods")
+			for _, pod := range pods {
+				if strings.Contains(pod.Name, "test-loadbalancer-ext") {
+					serverNodeName = pod.Node
+					break
+				}
+			}
+			return serverNodeName, nil
+		}, "25s", "5s").ShouldNot(BeEmpty(), "server pod not found")
+
+		var serverNodeIP string
+		for _, node := range nodes {
+			if node.Name == serverNodeName {
+				serverNodeIP = node.InternalIP
+			}
+		}
+
+		// Verify there is only one external-ip and it is matching the node IP
+		lbSvc := "nginx-loadbalancer-svc"
+		lbSvcExt := "nginx-loadbalancer-svc-ext"
+		Eventually(func() (int, error) {
+			externalIPs, _ := e2e.FetchExternalIPs(kubeConfigFile, lbSvc)
+			return len(externalIPs), nil
+		}, "25s", "5s").Should(Equal(2), "external IP count not equal to 2")
+
+		Eventually(func() (bool, error) {
+			externalIPs, _ := e2e.FetchExternalIPs(kubeConfigFile, lbSvcExt)
+			if len(externalIPs) != 1 {
+				return false, nil
+			}
+			return externalIPs[0] == serverNodeIP, nil
+		}, "25s", "5s").Should(BeTrue(), "external IP count not equal to 1 or external IP does not match serverNodeIP")
```

Comment:
```suggestion
		Eventually(func(g Gomega) (bool, error) {
			externalIPs, _ := e2e.FetchExternalIPs(kubeConfigFile, lbSvcExt)
			g.Expect(externalIPs).To(HaveLen(1), "more than 1 exernalIP found")
			g.Expect(externalIPs[0]).To(Equal(serverNodeIP),"external IP does not match servernodeIP")
		}, "25s", "5s").Should(Succeed())
```

Reply:
We can use the gomega [Assertions In Eventually](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode) feature to standardize the checks

Reply:
thanks! Your suggestion looks better :)